Octokit.Tests.Integration/Clients/Enterprise/EnterpriseManagementConsoleClientTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Threading.Tasks;
+using Octokit;
+using Octokit.Tests.Integration;
+using Xunit;
+
+public class EnterpriseManagementConsoleClientTests
+{
+    readonly IGitHubClient _github;
+
+    public EnterpriseManagementConsoleClientTests()
+    {
+        _github = EnterpriseHelper.GetAuthenticatedClient();
+    }
+
+    [GitHubEnterpriseTest]
+    public async Task CanGetMaintenanceMode()
+    {
+        var maintenance = await _github.Enterprise.ManagementConsole.GetMaintenanceMode("Password01");
+
+        Assert.NotNull(maintenance);
+    }
+
+    [GitHubEnterpriseTest]
+    public async Task CanSetMaintenanceModeOff()
+    {
+        // Set maintenance mode OFF now
+        var maintenance = await
+            _github.Enterprise.ManagementConsole.EditMaintenanceMode(
+                new UpdateMaintenanceRequest(),
+                EnterpriseHelper.ManagementConsolePassword);
+
+        Assert.NotNull(maintenance);
+        Assert.Equal(maintenance.Status, MaintenanceModeStatus.Off);
+    }
+
+    [GitHubEnterpriseTest]
+    public async Task CanSetMaintenanceModeOnNow()
+    {
+        // Set maintenance mode ON now
+        var maintenance = await
+            _github.Enterprise.ManagementConsole.EditMaintenanceMode(
+                new UpdateMaintenanceRequest(true, MaintenanceDate.Now()),
+                EnterpriseHelper.ManagementConsolePassword);
+
+        Assert.NotNull(maintenance);
+        Assert.Equal(maintenance.Status, MaintenanceModeStatus.On);
+
+        // Ensure maintenance mode is OFF
+        await _github.Enterprise.ManagementConsole.EditMaintenanceMode(
```

Comment:
From memory I wrote these tests to always return maintenance mode status back to a consistent state afterwards, so this was just ensuring matinenance mode was OFF after the test which turned it ON.  Or are you saying we should actually assert this here anyway?  Or potentially do it using a `Context` helper class, with the destructor implementing the "reset" logic?